tree/ntuple/v7/inc/ROOT/RNTupleDescriptor.hxx

```diff
@@ -506,6 +507,31 @@ public:
             fColumns.emplace_back(logicalId);
          }
       }
+
+      RColumnDescriptorIterable(const RNTupleDescriptor &ntuple)
+         : fNTuple(ntuple)
+      {
+         std::deque<DescriptorId_t> fieldIdQueue{ntuple.GetFieldZeroId()};
+
+         while (!fieldIdQueue.empty()) {
+            auto currFieldId = fieldIdQueue.front();
+            fieldIdQueue.pop_front();
+
+            for (const auto &field : ntuple.GetFieldIterable(currFieldId)) {
+               auto fieldId = field.GetId();
+
+               for (unsigned int i = 0; true; ++i) {
+                  auto logicalId = ntuple.FindLogicalColumnId(fieldId, i);
+                  if (logicalId == kInvalidDescriptorId)
+                     break;
+                  fColumns.emplace_back(logicalId);
+               }
```

Comment:
This block can be factored out in a separate method and then be used by both `RColumnDescriptorIterable` constructors.